### PR TITLE
[WebProfilerBundle] use the router to resolve file links

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #22290, #25886
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.xml [hunk 1]
```diff
@@ -56,7 +56,20 @@
...
+                        <argument type="service">
+                            <service class="string">
+                                <factory service="router" method="generate" />
+                                <argument>_profiler_open_file</argument>
+                            </service>
+                        </argument>
+                        <argument>?file=%%f&amp;line=%%l#line%%l</argument>
+                    </argument>
+                </service>
+            </argument>
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.xml]: this generates the following code, which is exactly what we need here:
```php
        return $this->privates['debug.file_link_formatter'] = new \Symfony\Component\HttpKernel\Debug\FileLinkFormatter(NULL, ($this->services['request_stack'] ?? $this->services['request_stack'] = new \Symfony\Component\HttpFoundation\RequestStack()), $this->targetDirs[3], \implode(array(0 => ($this->services['router'] ?? $this->getRouterService())->generate('_profiler_open_file'), 1 => '?file=%f&line=%l#line%l')));
```

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.xml [hunk 2]
```diff
@@ -56,7 +56,20 @@
             <argument>%debug.file_link_format%</argument>
             <argument type="service" id="request_stack" on-invalid="ignore" />
             <argument>%kernel.project_dir%</argument>
-            <argument>/_profiler/open?file=%%f&amp;line=%%l#line%%l</argument>
+            <argument type="service">
+                <service class="string">
```
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.xml]: nice hack!

File: src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.xml [hunk 3]
```diff
@@ -56,7 +56,20 @@
...
+                    <factory function="implode" />
```
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/config/profiler.xml]: oh, and yet another nice hack!
Well done :)

**Review by fabpot — APPROVED** [review]

**Comment by javiereguiluz** [comment]
@nicolas-grekas my head exploded when trying to understand this ... but I'm glad this is working 🎉 

However, the main concern of Fabien in the other PR (see https://github.com/symfony/symfony/pull/24153#discussion_r142044250) was the configurability of this. so, can you please paste an example of the config needed if the user wants to customize the URL? We'd need it to update the docs. Thanks!

**Comment by nicolas-grekas** [comment]
@javiereguiluz there is no config needed, the URL will be generated based on the path of the `_profiler_open_file` route. If you want another route, do it yourself (but that's not what the original issues asked for.)

**Review by xabbuh — APPROVED** [review]

**Review by javiereguiluz — APPROVED** [review]

**Comment by p5mtamazlicaru** [comment]
Also there is problem when doing the ```bin/console cache:clear --env=test``` have an error:
```cmd
In srcTestDebugProjectContainerUrlGenerator.php line 70:
                                                                                                    
  Unable to generate a URL for the named route "_profiler_open_file" as such route does not exist.
```

**Comment by thijskaspers** [comment]
Same here with `bin/console debug:router --env=ci`:

`In appCiDebugProjectContainerUrlGenerator.php line 81:
                                                                                                    
  Unable to generate a URL for the named route "_profiler_open_file" as such route does not exist.`

The 'ci' environment loads the `Symfony\Bundle\WebProfilerBundle\WebProfilerBundle()` in AppKernel.php so what I am doing wrong here?

**Comment by fabpot** [comment]
Shall we revert?

**Comment by nicolas-grekas** [comment]
It depends on the time we want, fix in progress, could be ready tomorrow.

**Comment by nicolas-grekas** [comment]
Fix in #26758

**Pull request merged**